app/services/CheckoutService.scala

```diff
@@ -276,7 +276,17 @@ class CheckoutService(identityService: IdentityService,
 
     def addPlan(contact: Contact) = {
       val newRatePlan = RatePlan(renewal.plan.id.get, None)
-      val renewCommand = Renew(subscription.id.get, subscription.termStartDate, subscription.termEndDate, NonEmptyList(newRatePlan), contractEffective, customerAcceptance)
+
+      val renewCommand = Renew(
+        subscriptionId = subscription.id.get,
+        currentTermStartDate = subscription.termStartDate,
+        currentTermEndDate = subscription.termEndDate,
+        newRatePlans = NonEmptyList(newRatePlan),
+        contractEffectiveDate = contractEffective,
+        customerAcceptanceDate = customerAcceptance,
+        promoCode = None,
+        autoRenew = renewal.plan.charges.billingPeriod.isInstanceOf[RecurringPeriod])
```

Comment:
maybe it's my java past but I always want to make autoRenew a def in the Billing period trait instead of asking if something implements a trait like. But I guess this is the same as doing pattern matching, thoughts?